### PR TITLE
[AppService] Fix #5720946: az webapp backup fails to set name

### PR DESCRIPTION
**Pull request by raohri03**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by pvbchak** [comment]
LGTM

**Comment by yonzhan** [comment]
add to S164.

**Comment by Juliehzl** [comment]
It looks that there is no test for `az webapp config backup` command. Can you add test for this command to help us find SDK changes through test?

**Comment by panchagnula** [comment]
@raohri03 can you add a test for this please? Is there an existing issue reported here? Thanks!

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1470,7 +1470,7 @@ def create_backup(cmd, resource_group_name, webapp_name, storage_account_url,
     if backup_name and backup_name.lower().endswith('.zip'):
         backup_name = backup_name[:-4]
     db_setting = _create_db_setting(cmd, db_name, db_type=db_type, db_connection_string=db_connection_string)
-    backup_request = BackupRequest(backup_request_name=backup_name,
+    backup_request = BackupRequest(backup_name=backup_name,
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Why you change this line? Is it a mistake before? or SDK has a change?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Yes, there is a SDK change.

**Comment by raohri03** [comment]
@panchagnula will add test today, apologies for delayed response

**Comment by Juliehzl** [comment]
@raohri03 any update?

**Comment by raohri03** [comment]
Can you please approve this  for now we will open a new work item and write tests for whole backup restore

**Comment by panchagnula** [comment]
> Can you please approve this for now we will open a new work item and write tests for whole backup restore

@raohri03 the problem is we don't understand the purpose of this change - test will help us understand the purpose of this change. Is there a reason you want this to be merged? is there an existing bug you are trying to fix?

**Comment by pvbchak** [comment]
> > Can you please approve this for now we will open a new work item and write tests for whole backup restore
> 
> @raohri03 the problem is we don't understand the purpose of this change - test will help us understand the purpose of this change. Is there a reason you want this to be merged? is there an existing bug you are trying to fix?

@panchagnula , Fix is to use correct field in request payload (Corresponding Rest API: https://docs.microsoft.com/en-us/rest/api/appservice/webapps/backup). Site backup succeeds even if this field is not set (backend generates a name with timestamp suffix when this field is not set). With fix, backend will honor the name set in CLI while generating a backup.

**Comment by yonzhan** [comment]
How is the status of this PR ?

**Pull request merged**